paper/src/main/java/com/denizenscript/denizen/paper/events/WorldGameRuleChangeScriptEvent.java

```diff
@@ -0,0 +1,144 @@
+    //
+    // @Context
+    // <context.gamerule> returns name of the GameRule which was changed.
+    // <context.value> returns the new value of the GameRule.
+    // <context.world> returns the world where the GameRule is applied.
+    // <context.source_type> returns the sender who changed the GameRule.
+    // <context.command_block_location> returns the command block's location (if the command was run from one).
+    // <context.command_minecart> returns the EntityTag of the command minecart (if the command was run from one).
+    //
+    // @Determine
+    // "VALUE:" + ElementTag(Number) or ElementTag(Boolean) to set the value of the GameRule.
+    //
+    // @Player when the sender of the command is a player.
+    //
+    // -->
+
+    public WorldGameRuleChangeScriptEvent() {
+        registerCouldMatcher("gamerule changes (in <world>)");
+    }
+
+    public WorldGameRuleChangeEvent event;
+
+    public WorldTag world;
+    public LocationTag commandBlockLocation;
+    public EntityTag commandMinecart;
+    public ElementTag sourceType;
+    public CommandSender source;
+
+    @Override
+    public boolean matches(ScriptPath path) {
+        if (path.eventArgLowerAt(2).equals("in") && !world.tryAdvancedMatcher(path.eventArgLowerAt(3))) {
+            return false;
+        }
+        return super.matches(path);
+    }
+
+    @Override
+    public ObjectTag getContext(String name) {
+        switch (name) {
+            case "gamerule": return new ElementTag(event.getGameRule().getName());
+            case "value": return new ElementTag(event.getValue());
+            case "source_type": return sourceType;
+            case "command_block_location": return commandBlockLocation;
+            case "command_minecart": return commandMinecart;
+            case "world": return world;
+        }
+        return super.getContext(name);
+    }
+
+    @Override
+    public boolean applyDetermination(ScriptPath path, ObjectTag determinationObj) {
+        if (determinationObj instanceof ElementTag) {
+            String lower = CoreUtilities.toLowerCase(determinationObj.toString());
+            if (lower.startsWith("value:")) {
+                ElementTag value = new ElementTag(lower.substring("value:".length()));
+                if (value.isInt() || value.isBoolean()) {
+                    event.setValue(value.toString());
+                    return true;
+                }
+            }
+        }
+        return super.applyDetermination(path, determinationObj);
+    }
+
+    @Override
+    public ScriptEntryData getScriptEntryData() {
+        PlayerTag player = null;
+        if (source instanceof Player) {
+            player = new PlayerTag((Player) source);
+        }
+        return new BukkitScriptEntryData(player, null);
+    }
+
+    @EventHandler
+    public void onGameRuleChanged(WorldGameRuleChangeEvent event) {
+        this.source = event.getCommandSender();
+        this.commandMinecart = getCommandMinecart();

```

Comment:
This values don't need to be pre-loaded, they can be gotten from the `getContext` method

Reply:
Moved sourceType, commandMinecart and commandBlockLocation to getContext.